app/measurement/management/commands/s3_query_export.py

```diff
@@ -0,0 +1,217 @@
+                curr_date = start_date + timedelta(days=kday)
+                # used for SQL query
+                days_before_today = (date.today() - curr_date).days
+
+                # Get all metrics available for this date
+                metrics = self.get_db_date_metrics(cursor, curr_date)
+
+                # Select single metric if requested
+                if kwargs['metric'] >= 0:
+                    if kwargs['metric'] in metrics:
+                        metrics = [kwargs['metric']]
+                    else:
+                        metrics = []
+
+                for metric in metrics:
+                    '''
+                    file name format:
+                    YYYY/mm_dd_metric_{id}.csv
+                    '''
+                    file_path = (
+                        f"{curr_date.strftime('%Y/%m_%d')}_metric_{metric}.csv"
+                    )
+                    print(f"save to {file_path}")
+
+                    # Check overwrite status, does metric file already exist?
+                    if not kwargs['overwrite']:
+                        if self.check_s3_file_exists(file_path):
+                            print("Not writing file either because it exists"
+                                  " or cannot confirm its existence!")
+                            continue
+
+                    cursor.execute(sql, [
+                        metric,
+                        days_before_today,
+                        days_before_today - 1,
+                        self.BUCKET_NAME,
+                        file_path,
+                        self.REGION
+                    ])
+
+    def get_db_date_metrics(self, cursor, check_date):
+        '''
+        Get all metrics for a given date.
+        '''
+        sql = '''
+            SELECT DISTINCT metric_id FROM measurement_measurement
+            WHERE starttime >= current_date - %s
+                AND starttime < current_date - %s
+            ORDER BY metric_id;
+        '''
+        days_before_today = (date.today() - check_date).days
+        cursor.execute(sql, [days_before_today, days_before_today - 1])
+        return [res[0] for res in cursor.fetchall()]
+
+    def check_s3_file_exists(self, file_path):
+        '''
+        Check if file is in the s3 bucket. AWS_ACCESS_KEY_ID and
+        AWS_SECRET_ACCESS_KEY should be defined in .env (or elsewhere?) for
+        this to work.
+        Default return value is False in case there are s3 access errors
+        '''
+        try:
+            s3 = boto3.resource('s3')
+            bucket = s3.Bucket(self.BUCKET_NAME)
+            objs = list(bucket.objects.filter(Prefix=file_path))
+        except botocore.exceptions.NoCredentialsError:
+            print('AWS credentials do not exist!')
+            return False
+        except botocore.exceptions.ClientError:
+            print('Cannot connect to aws client, check your credentials!')
+            return False
+
+        return len(objs) == 1
```

Comment:
I think this should be  
`returnlen(objs) !=0`

to cover the case of duplicates already existing. I don't think this is possible but you never know.